lte/gateway/c/core/oai/tasks/amf/amf_app_handler.cpp

```diff
@@ -104,9 +104,12 @@ void amf_ue_context_update_coll_keys(amf_ue_context_t* const amf_ue_context_p,
                  amf_ue_ngap_id);
   }
 
-  if ((amf_ue_ngap_id != INVALID_AMF_UE_NGAP_ID) && (imsi != 0)) {
-    m_rc = amf_ue_context_p->imsi_amf_ue_id_htbl.remove(
-        ue_context_p->amf_context.imsi64);
+  m_rc = amf_ue_context_p->imsi_amf_ue_id_htbl.remove(
```

Comment:
Removal of imsi from imsi_amf_ue_id_htbl, should be under below condition, because if we update gnb_ngap_id_key, we don't return from there so we will hit this statement and removes the entry from hash list.

Reply:
m_rc is not required here

Reply:
Comment addressed.